src/app/audiologist-navigation/navigation-aids.ts

```diff
@@ -0,0 +1,104 @@
+import { Utilities } from '../common/utlilities';
+
+/*
+    If you need to add a new option to the audiologist and/or admin:
+    1. Add a new item in TabsEnum
+    2. Add to setupState() under the appropriate state(s) it should appear
+*/
+export enum TabsEnum {
+    SUMMARY,
+    TESTS,
+    NOTES,
+    SEARCH,
+    ACCOUNT,
+    USERS,
+    QUERIES,
+    SPREAD,
+    SUBMIT_BTN,
+    DISCARD_BTN,
+    CLOSE_BTN
+};
+export enum StatesEnum {
+    FROM_QUEST,
+    AUD_NO_DATA,
+    ADMIN_NO_DATA,
+    LOADED_APPT
+}
+
+export class State {
+    private tabAvailable: Map<TabsEnum, boolean> = new Map();
+    private currentTab: TabsEnum;
+
+    constructor() {
+        this.determineState();
+    }
+
+    public tabIsAvailable(tab: TabsEnum): boolean {
+        return this.tabAvailable.get(tab);
+    }
+
+    public tabIsSelected(tab: TabsEnum): boolean {
+        return tab === this.currentTab;
+    }
+
+    public selectTab(tab: TabsEnum) {
+        this.currentTab = tab;
+    }
+
+    public determineState() {
+        if (Utilities.getSessionStorage('dataFromDB') === 'true') {
+            this.setupState(StatesEnum.LOADED_APPT);
+        } else if (Utilities.getSessionStorage('dataFromDB') === 'false') {
+            this.setupState(StatesEnum.FROM_QUEST);
+        } else {
```

Comment:
It only gets here if `dataFromDB` is neither true nor false; is that correct? Does that handle the case of using this navigation multiple times?

Reply:
No more weird dataFromDB in multiple files.  Now just takes two booleans to determine what state it's in.
Audiologist-navigation tells it if there is appointment data, and where it came from if any.